test/client/storage.test.ts

```diff
@@ -100,16 +103,12 @@ describe("StateMemoryEngine", () => {
             }
             await smallTreeEngine.commit();
 
-            let errMsg = "";
-            try {
-                await smallTreeEngine.findVacantSubtree(maxSubtreeDepth);
-            } catch (err) {
-                errMsg = err.message;
-            }
-            assert.equal(
-                errMsg,
-                `Tree at level ${depth -
-                    maxSubtreeDepth} is full, no room for subtree insert`
+            assert.isRejected(
+                smallTreeEngine.findVacantSubtree(maxSubtreeDepth),
+                new RegExp(
```

Comment:
The regex match looks fragile to the change of the error message.
What do you think about creating a custom error class `TreeAtLevelIsFull` like what we did in the `exceptions.ts`, so we can check against `error.name` and `error.level`?